core/src/View/MDSpan/Kokkos_MDSpan_Header.hpp

```diff
@@ -44,7 +44,39 @@ using std::layout_stride;
 using std::mdspan;
 }  // namespace Kokkos
 #else
+// Opt in for Kokkos::pair to submdspan/subview
+// submdspan does only take index_pair_like which is derived from tuple_like
+// tuple_like is an enumerated list: tuple, pair, array, complex,
+// ranges::subrange Needs to be defined before including mdspan header
+
+#include <Kokkos_Pair.hpp>
+
+namespace Kokkos {
+namespace detail {
+template <class IdxT1, class IdxT2>
+KOKKOS_INLINE_FUNCTION constexpr auto first_of(
+    const pair<IdxT1, IdxT2> &slice) {
+  return slice.first;
+}
+template <class IdxT1, class IdxT2, class Extents, size_t k>
+KOKKOS_INLINE_FUNCTION constexpr auto last_of(std::integral_constant<size_t, k>,
+                                              const Extents &,
+                                              const pair<IdxT1, IdxT2> &slice) {
+  return slice.second;
+}
+
+template <class T, class IndexType>
+struct index_pair_like;
+
+template <class IdxT1, class IdxT2, class IndexType>
+struct index_pair_like<Kokkos::pair<IdxT1, IdxT2>, IndexType> {
+  static constexpr bool value = std::is_convertible_v<IdxT1, IndexType> &&
+                                std::is_convertible_v<IdxT2, IndexType>;
+};
+}  // namespace detail
+}  // namespace Kokkos
 #include <mdspan/mdspan.hpp>
```

Comment:
Moving the `include` up should work since we are just specializing templates declared in `mdspan`/`mdspan.hpp`, right?

Reply:
I thought I had issues because the submdspan implementation needs to use this specialization.

Reply:
Yes, I tested it and it needs to be here since I suppose the template is instantiated somewhere in the mdspan header